### PR TITLE
24205 - updates for name change and alteration

### PR DESCRIPTION
**Pull request by ketaki-deodhar**
*Issue #:* /bcgov/entity#24205

*Description of changes:*

- Remove FFs `supported-name-change-entities` and `supported-alteration-types`
- If business is in LEAR, go to modernized system, otherwise, go to COLIN.

**Name Change:**
Business in LEAR:

image

Not in LEAR:
image

**Manage My Name Request:**

In LEAR: Navigates user to Dashboard
image
image

Not in LEAR:
Navigates user to Corp Online
image

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the namerequest license (Apache 2.0).

### CODE REVIEW

File: app/src/store/state.ts
Comment (severinbeauvais) [inline app/src/store/state.ts:82]: Does this need to get reset if the user aborts the current name request flow and starts anew?

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://namerequest-dev--pr-784-ek34zebb.web.app

File: app/src/mixins/search-mixin.ts
Comment (severinbeauvais) [inline app/src/mixins/search-mixin.ts:38]: GHA warning says that BusinessSearchIF no longer needs to be imported.

Comment (ketaki-deodhar) [reply app/src/mixins/search-mixin.ts:38]: oops missed removing! Done

File: app/src/components/new-request/search-components/business-lookup-fetch.vue
Comment (severinbeauvais) [inline app/src/components/new-request/search-components/business-lookup-fetch.vue:122]: You don't need conditional chaining because the object is already validated on line 118.

Conditional chaining is a double-edged sword as it can mask errors. Not here, but generally we need to be careful with it, and here you don't need it.

Comment (ketaki-deodhar) [reply app/src/components/new-request/search-components/business-lookup-fetch.vue:122]: Yes! Removing

File: app/src/components/existing-request/nr-approved-gray-box.vue
Comment (severinbeauvais) [inline app/src/components/existing-request/nr-approved-gray-box.vue:304]: This is an async function and needs to be awaited.

File: app/src/store/state.ts
Comment (ketaki-deodhar) [reply app/src/store/state.ts:82]: It is calculated each time a new business is entered

File: app/src/mixins/common-mixin.ts
Comment (severinbeauvais) [inline app/src/mixins/common-mixin.ts:249]: For the record (no need to change this), another way to do this is to have it as an async store action.

**Review by severinbeauvais — COMMENTED** [review]
I tested a CHG NR. Looks good. I'll approve after the await changes.

File: app/src/components/new-request/search-components/business-lookup-fetch.vue
Comment (ketaki-deodhar) [reply app/src/components/new-request/search-components/business-lookup-fetch.vue]: Done

File: app/src/components/existing-request/nr-approved-gray-box.vue
Comment (ketaki-deodhar) [inline app/src/components/existing-request/nr-approved-gray-box.vue:313]: separated the logic for async call instead of making a 'get' async function.

File: app/src/components/existing-request/nr-approved-gray-box.vue
Comment (severinbeauvais) [inline app/src/components/existing-request/nr-approved-gray-box.vue:348]: Why not just await this in mounted() (and make that async to match)?

File: app/src/components/existing-request/nr-approved-gray-box.vue
Comment (severinbeauvais) [inline app/src/components/existing-request/nr-approved-gray-box.vue:251]: Technically this won't work because it's async and you're not awaiting it. In other words, mounted() will complete before checkBusiness() is done.

Just FYI since my other comment proposed a cleanup that will obsolete this comment.

File: app/src/components/existing-request/nr-approved-gray-box.vue
Comment (ketaki-deodhar) [reply app/src/components/existing-request/nr-approved-gray-box.vue]: Yes. I think that is better.

Comment (severinbeauvais) [reply app/src/components/existing-request/nr-approved-gray-box.vue]: Beauty! Simple is best. Resolving.

**Review by severinbeauvais — APPROVED** [review]
LGTM. Poke Megan for a second review.

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://namerequest-dev--pr-784-ek34zebb.web.app

**Review by meawong — APPROVED** [review]
tested and LGTM

**Pull request merged**
